opentelemetry-otlp/src/exporter/grpcio.rs

```diff
@@ -85,7 +90,9 @@ impl GrpcioExporterBuilder {
 
     /// Set additional headers to send to the collector.
     pub fn with_headers(mut self, headers: HashMap<String, String>) -> Self {
-        self.grpcio_config.headers = Some(headers);
+        let mut inst_headers = self.grpcio_config.headers.unwrap_or_default();
+        inst_headers.extend(headers.into_iter());
+        self.grpcio_config.headers = Some(inst_headers);
```

Comment:
nit: is it possible to use https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert here? It may looks cleaner

Reply:
I'm not sure if that works -- since what we need is the headers provided PLUS the headers already found in `self.grpcio_config.headers` -- `get_or_insert` would only allow us to overwrite the headers, OR write over them with `headers` provided.

Reply:
so we'd run into a couple situations (for future proofing)
1. We'd be writing over the headers we had already put in (User-Agent plus whatever we may include later)
2. We would get a pointer back, which we then have to do the same work to write into and put back into it's own.. reference? Getting the ref might be tricky to update that data OR write over it, I guess.

Maybe I'm missing something with your suggestion -- but it seems like `get_or_insert` would write over the non-existent headers or we'd have to extend this way anyways. What do you think?

Reply:
Emm I assumed your intention is to override the `User-Agent` header even when the user already provided one. In that case, `get_or_insert` can be used. Note that the `extend` method can be applied to a mutable reference and it will just change the Hashmap the reference points to. Here is an [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d5f0a91fd1aedeb9e6a19ab891a61279).

On second thought I think we should not override the `User-Agent` header if the user provided one. So[ `extend` ](https://doc.rust-lang.org/std/collections/struct.HashMap.html#impl-Extend%3C(K%2C%20V)%3E-for-HashMap%3CK%2C%20V%2C%20S%3E) probably won't work in this case as it will override the value. We can still get a mutable reference using `get_or_insert`, we can then check if the `User-Agent` is present and add one if it doesn't

Reply:
@GaryPWhite Let me know if you think this makes sense or if maybe I misunderstand something here?

Reply:
put in get_or_insert